File: ios/Sources/NativeAudioPlugin/Constant.swift
//
//  Constant.swift
//  Plugin
//
//  Created by priyankpat on 2020-05-28.
//  Copyright © 2022 Martin Donadieu. All rights reserved.
//

public class Constant {
    // Parameter keys
    public static let FadeKey = "fade"
    public static let FocusAudio = "focus"
    public static let AssetPathKey = "assetPath"
    public static let AssetIdKey = "assetId"
    public static let Volume = "volume"
    public static let Rate = "rate"
    public static let Loop = "loop"
    public static let Background = "background"
    public static let IgnoreSilent = "ignoreSilent"
    public static let ShowNotification = "showNotification"
    public static let NotificationMetadata = "notificationMetadata"

    // Default values - used for consistency across the plugin
    public static let DefaultVolume: Float = 1.0
    public static let DefaultRate: Float = 1.0
    public static let DefaultChannels: Int = 1
    public static let DefaultFadeDelay: Float = 1.0
    public static let MinRate: Float = 0.25
    public static let MaxRate: Float = 4.0
    public static let MinVolume: Float = 0.0
    public static let MaxVolume: Float = 1.0
    public static let MaxChannels: Int = 32

    // Error messages
    public static let ErrorAssetId = "Asset Id is missing"
    public static let ErrorAssetPath = "Asset Path is missing"
    public static let ErrorAssetNotFound = "Asset is not loaded"
    public static let ErrorAssetAlreadyLoaded = "Asset is already loaded"
}

File: android/src/main/java/ee/forgr/audio/Constant.java
package ee.forgr.audio;

public class Constant {

    public static final String ERROR_AUDIO_ID_MISSING = "Audio Id is missing";
    public static final String ERROR_AUDIO_ASSET_MISSING = "Audio Asset is missing";
    public static final String ERROR_AUDIO_EXISTS = "Audio Asset already exists";
    public static final String ERROR_ASSET_PATH_MISSING = "Asset Path is missing";
    public static final String ERROR_ASSET_NOT_LOADED = "Asset is not loaded";

    public static final String ASSET_ID = "assetId";
    public static final String ASSET_PATH = "assetPath";
    public static final String OPT_FADE_MUSIC = "fade";
    public static final String OPT_FOCUS_AUDIO = "focus";
    public static final String VOLUME = "volume";
    public static final String RATE = "rate";
    public static final String AUDIO_CHANNEL_NUM = "audioChannelNum";
    public static final String LOOP = "loop";
    public static final String SHOW_NOTIFICATION = "showNotification";
    public static final String NOTIFICATION_METADATA = "notificationMetadata";
    public static final int INVALID = 0;
    public static final int PREPARED = 1;
    public static final int PENDING_PLAY = 2;
    public static final int PLAYING = 3;
    public static final int PENDING_LOOP = 4;
    public static final int LOOPING = 5;
    public static final int PAUSE = 6;
}
